src/main/java/org/eclipse/yasson/internal/serializer/DateTypeSerializer.java

```diff
@@ -35,7 +37,42 @@ public DateTypeSerializer(Customization customization) {
         super(customization);
     }
 
-    protected DateTimeFormatter getDefaultFormatter() {
-        return DEFAULT_FORMATTER;
+    @Override
+    protected Instant toInstant(Date value) {
+        if (value instanceof java.sql.Date) {
```

Comment:
This will affect performance a bit. I'd better keep serializers for each of the types as a separate classes. Each of them will contain half of this code and you do not have to test what target class it is.

Reply:
We have to do the instanceof check here because users have done things like this:
```java
public class Foo {
  java.util.Date date = new java.sql.Date(...);
}
```
Since the property type is `java.util.Date` the java.util.Date serializer will get called, even though the runtime type of the value is java.sql.Date. See this issue for more details: https://github.com/eclipse-ee4j/yasson/issues/68